storops/unity/resource/system.py

```diff
@@ -124,12 +124,26 @@ def get_snap(self, _id=None, name=None, **filters):
     def create_nas_server(self, name, sp=None, pool=None, is_repl_dst=None,
                           multi_proto=None, tenant=None):
         if sp is None:
-            sp = self.get_sp().first_item
+            sp = self._auto_balance_sp()
 
         return sp.create_nas_server(name, pool,
                                     is_repl_dst=is_repl_dst,
                                     multi_proto=multi_proto, tenant=tenant)
 
+    def _auto_balance_sp(self):
+        sp_list = self.get_sp()
+        if len(sp_list) < 2:
+            LOG.debug('spb not present. Pick spa to host nas server.')
+            return sp_list.first_item
+
+        servers = self.get_nas_server()
+        spa, spb = sp_list
+        servers_on_spa = servers.shadow_copy(home_sp=spa)
+        sp = spb if len(servers_on_spa) * 2 > len(servers) else spa
+        LOG.debug('Pick %s to balance of spa and spb to host nas servers.',
+                  sp.get_id())
+        return sp

```

Comment:
No need to change but keep in mind.  The first character of the sentence need to be lower case as well.